### PR TITLE
#3071: create new directory setting and use for '-d' parameter

### PR DESCRIPTION
**Pull request by esarbe**
also make sure that directory exists when updating setting

### CODE REVIEW

File: compiler/src/dotty/tools/dotc/config/Settings.scala [hunk 1]
```diff
@@ -158,6 +161,10 @@ object Settings {
             case Success(v) => update(v, args)
             case Failure(ex) => fail(ex.getMessage, args)
           }
+        case (DirectoryTag, arg :: args) =>
+          val path = Path(arg)
+          if (path.isDirectory) update(arg, args)
```
Comment (allanrenucci) [inline compiler/src/dotty/tools/dotc/config/Settings.scala]: Here, it should be `update(path.toDirectory, args)`

File: compiler/src/dotty/tools/dotc/config/ScalaSettings.scala [hunk 2]
```diff
@@ -17,7 +18,7 @@ class ScalaSettings extends Settings.SettingGroup {
   val javaextdirs = PathSetting("-javaextdirs", "Override java extdirs classpath.", Defaults.javaExtDirs)
   val sourcepath = PathSetting("-sourcepath", "Specify location(s) of source files.", "") // Defaults.scalaSourcePath
   val classpath = PathSetting("-classpath", "Specify where to find user class files.", defaultClasspath) withAbbreviation "-cp"
-  val d = StringSetting("-d", "directory|jar", "destination for generated classfiles.", ".")
+  val d = DirectorySetting("-d", "directory|jar", "destination for generated classfiles.", Directory(Path(".")))
```
Comment (allanrenucci) [inline compiler/src/dotty/tools/dotc/config/ScalaSettings.scala]: We should probably rename this setting to something like `outputDir`

**Comment by esarbe** [comment]
I've had to touch up the tests so that the output directory actually exists.

**Comment by allanrenucci** [comment]
Could you add a test under `compiler/test/dotty/tool/dotc/SettingTests.scala` and do something like:
```scala
class SettingTests {
  ...

  @Test def missingOutputDir: Unit = {
    val options = Array("-d", "not_here")
    val reporter = Main.process(options)
    assertEquals(reporter.errorCount, 1)
    assertEquals(reporter.allErrors.head.message, "'not_here' does not exist or is not a directory")
  }
}
```

**Comment by esarbe** [comment]
Sure thing!

**Comment by smarter** [comment]
Nitpick: To automatically close the issue when a PR is merged, you should put "Fix #nnnn" in the commit message, just putting "#nnnn" is not enough.

**Comment by esarbe** [comment]
Thanks for the nitpicking! I've updated the last commit comment, I hope that suffices.

**Pull request merged**